src/main/java/frc/robot/commands/autos/components/DriveDistance.java

```diff
@@ -0,0 +1,60 @@
+package frc.robot.commands.autos.components;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.RobotContainer.MatchState;
+import frc.robot.subsystems.drive.DriveSubsystem;
+
+/** drives in a direction at a given speed for a certain distance */
+public class DriveDistance extends SequentialCommandGroup {
+  double distanceTraveledMeters = 0;
+  
+  Pose2d startPose;
+  Translation2d direction;
+  Translation2d speeds;
+  double distanceToTravel;
+  MatchState matchState;
+
+  /**
+   * @param normSpeed should be in [0,1]
+   */
+  public DriveDistance(
+      DriveSubsystem drive,
+      double normSpeed,
+      double XMeters,
+      double YMeters,
```

Comment:
Need a comment that this XY is based on the 2023 coordinate system (positive X is away from driver station no matter the alliance)